pa_config/rca.conf

```diff
@@ -86,11 +86,21 @@
     "ClusterTemperatureRca"
   ],
 
+  // Default Values are added in the config specific file as well
+  // Modify them as well whenever changing the default values (DeciderConfig.java)
   "decider-config-settings": {
+    // Decreasing order of priority for the type of workload we can expect on the cluster.
+    "workload-type": {
+      "priority-order": ["ingest", "search"]
+    },
+    // Decreasing order of priority for the type of cache which is expected to be consumed more.
+    "cache-type": {
+      "priority-order": ["read-heavy", "write-heavy"]
```

Comment:
a high level question is how do we plan to use this cache priority in decider. I understand the purpose of having cache is to accelerate reading. So what should decider do if write heavy is specified here?  
And if customer wants to prioritized fielddata cache over shard request cache, how should they set it in config  ?

Reply:
The assumption here is customer doesn't know in detail about the field data cache and the shard request cache. So customer is going to provide the inputs in terms of there workload priority and cache type priority. Our deciders then in turn would change these variables into the prioritising Field data cache over shard request cache.

Reply:
Caches are different from queues and don't easily map to workload types. From what i know, all caches help with some form of read traffic in ES. Indexing has a separate indexing buffer.
For caches, customers will have to call out the priority wrt. specific cache names - `field-data-cache, shard-request-cache` etc. A read/write heavy cluster does not provide data on relative cache priorities. 
These caches are public documentation so customers would know about it, but it would still be an advanced setting.

Reply:
So do you suggest we should remove the read-heavy / write-heavy input altogether? If we don't want to use this for cache then the other things we needed it for can be derived from workload (search/ingest) type?

Reply:
My suggestion is to directly input cache priorities, i.e. relevant caches for the customer from most imp to least imp (or vice versa). You can think of what the config key should be, but values will contain actual cache names. 

Something like - 
```ruby
"cache-priority": ["fielddata-cache", "shard-request-cache", "query-cache", "bitset-filter-cache"]
```

We cannot derive this from workload (search/ingest) types.

Reply:
Right! I have updated the default values to specific caches. The output we wanted to derive from read-heavy / write-heavy can be taken from search / ingest as well but we do need the explicit cache priority.